MekHQ/src/mekhq/gui/adapter/UnitTableMouseAdapter.java

```diff
@@ -526,7 +520,25 @@ public void actionPerformed(ActionEvent action) {
                             part.setTeamId(null);
                             part.cancelReservation();
                         }
+                        
+                        // replace damaged armor and reload ammo bins after fixing their respective locations
+                        if(part instanceof Armor) {
+                            final Armor armor = (Armor) part;
+                            armor.setAmount(armor.getTotalAmount());
+                        } else if(part instanceof AmmoBin) {
+                            final AmmoBin ammoBin = (AmmoBin) part;
+                            
+                            // we magically find the ammo we need, then load the bin
+                            // we only want to get the amount of ammo the bin actually needs                            
+                            if(ammoBin.getShotsNeeded() > 0) {
+                                ammoBin.getAcquisitionWork(ammoBin.getShotsNeeded()).find(0);
```

Comment:
`AmmoStorage.find(int)` calls Campaign.buyPart(Part, int)`, which debits the cost of the ammo from the finances (unless paying for parts is disabled). This is not what I would expect with a GM restore.

I tested with this and it appeared successful:
```                            
if (ammoBin.getShotsNeeded() > 0) {
    ammoBin.setShotsNeeded(0);
    ammoBin.updateConditionFromPart();
}
```